packages/preferences/README.md

```diff
@@ -0,0 +1,174 @@
+# Preferences
+
+Utilities for storing WordPress preferences.
+
+## Installation
+
+Install the module
+
+```bash
+npm install @wordpress/preferences --save
+```
+
+_This package assumes that your code will run in an **ES2015+** environment. If you're using an environment that has limited or no support for such language features and APIs, you should include [the polyfill shipped in `@wordpress/babel-preset-default`](https://github.com/WordPress/gutenberg/tree/HEAD/packages/babel-preset-default#polyfill) in your code._
+
+## Examples
+
+### Data store
+
+Preferences are persisted values of any kind.
+
+Set the default preferences for any features on initialization by dispatching an action:
+
+```js
+import { dispatch } from '@wordpress/data';
+import { store as preferencesStore } from '@wordpress/preferences';
+
+function initialize() {
+	// ...
+
+	dispatch( preferencesStore ).setDefaults(
+		'namespace/editor-or-plugin-name',
+		{
+			myBooleanFeature: true,
+		}
+	);
+
+	// ...
+}
+```
+
+Or the `get` selector to get a preference value, and the `set` action to update a preference to any value:
+
+```js
+wp.data
+	.select( 'core/preferences' )
+	.get( 'namespace/editor-or-plugin-name', 'myPreferenceName' ); // 1
+wp.data
+	.dispatch( 'core/preferences' )
+	.set( 'namespace/editor-or-plugin-name', 'myPreferenceName', 2 );
+wp.data
+	.select( 'core/preferences' )
+	.get( 'namespace/editor-or-plugin-name', 'myPreferenceName' ); // 2
+```
+
+Use the `toggle` action to flip a boolean preference between `true` and `false`:
+
+```js
+wp.data
+	.select( 'core/preferences' )
+	.get( 'namespace/editor-or-plugin-name', 'myPreferenceName' ); // true
+wp.data
+	.dispatch( 'core/preferences' )
+	.toggle( 'namespace/editor-or-plugin-name', 'myPreferenceName' );
+wp.data
+	.select( 'core/preferences' )
+	.get( 'namespace/editor-or-plugin-name', 'myPreferenceName' ); // false
+```
+
+### Components
+
+The `PreferenceToggleMenuItem` components can be used with a `DropdownMenu` to implement a menu for changing preferences.
+
+Also see the `MoreMenuDropdown` component from the `@wordpress/interface` package for implementing a more menu.
+
+```jsx
+function MyEditorMenu() {
+	return (
+		<MoreMenuDropdown>
+			{ () => (
+				<MenuGroup label={ __( 'Features' ) }>
+					<PreferenceToggleMenuItem
+						scope="namespace/editor-or-plugin-name"
+						name="myPreferenceName"
+						label={ __( 'My feature' ) }
+						info={ __( 'A really awesome feature' ) }
+						messageActivated={ __( 'My feature activated' ) }
+						messageDeactivated={ __( 'My feature deactivated' ) }
```

Comment:
I think this is not new to this PR but in other components that needed configurable labels, we had a single object prop like `messages` in `FormTokenField`. Feels like something we'd need to normalize at some point.